bundles/org.openhab.binding.wundergroundupdatereceiver/src/main/java/org/openhab/binding/wundergroundupdatereceiver/internal/WundergroundUpdateReceiverServlet.java

```diff
@@ -0,0 +1,183 @@
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.wundergroundupdatereceiver.internal;
+
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toSet;
+import static org.openhab.binding.wundergroundupdatereceiver.internal.WundergroundUpdateReceiverBindingConstants.*;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.io.http.servlet.BaseOpenHABServlet;
+import org.osgi.service.http.HttpContext;
+import org.osgi.service.http.HttpService;
+import org.osgi.service.http.NamespaceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WundergroundUpdateReceiverServlet} is responsible for receiving updates,and
+ * updating the matching channels.
+ *
+ * @author Daniel Demus - Initial contribution
+ */
+@NonNullByDefault
+public class WundergroundUpdateReceiverServlet extends BaseOpenHABServlet {
+
+    public static final String SERVLET_URL = "/weatherstation/updateweatherstation.php";
+    private static final long serialVersionUID = -5296703727081438023L;
+
+    private final Logger logger = LoggerFactory.getLogger(WundergroundUpdateReceiverServlet.class);
+    private final Map<String, WundergroundUpdateReceiverHandler> handlers = new HashMap<>();
+
+    private static final Object LOCK = new Object();
+
+    private boolean active = false;
+
+    public WundergroundUpdateReceiverServlet(HttpService httpService) {
+        super(httpService);
+    }
+
+    public boolean isActive() {
+        synchronized (LOCK) {
+            return this.active;
+        }
+    }
+
+    public Set<String> getStationIds() {
+        return this.handlers.keySet();
+    }
+
+    public void activate() {
+        activate(SERVLET_URL, httpService.createDefaultHttpContext());
+    }
+
+    @Override
+    protected void activate(String alias, HttpContext httpContext) {
+        synchronized (LOCK) {
+            try {
+                logger.debug("Starting servlet {} at {}", getClass().getSimpleName(), alias);
+                Dictionary<String, String> props = new Hashtable<>(1, 10);
+                httpService.registerServlet(alias, this, props, httpContext);
+                active = true;
+            } catch (NamespaceException e) {
+                logger.error("Error during servlet registration - alias {} already in use", alias, e);
```

Comment:
error level should only be for critical errors. Use warn level if it is the correct level after reading here...
https://www.openhab.org/docs/developer/guidelines.html#f-logging
Please go over all uses of the logger.

Reply:
But this should never happen unless 1: another binding takes the servlet alias, which is showstopping problem, or 2: somehow two instances of the binding are instantiated which is also a showstopper.

Reply:
If it stops the openHAB core from working then it should be error, if it is not critical to the openhab core then it should be a warn. If the binding cant be used then do not log at all and place the thing to OFFLINE.